app/ui/src/app/integration/integration_detail/integration-description.component.html

```diff
@@ -25,7 +28,7 @@
         <span *ngIf="!last" class="step-sep fa fa-angle-right"></span>
       </div>
     </ng-template>
-  </p>
+  </div>
```

Comment:
You can make all this block leaner by removing redundant structural container directives:

```html
  <div *ngFor="let step of integration?.steps; let index=index, last=last" class="step-block inline-block text-center">
    <ng-container [ngSwitch]="step.stepKind">
      <div *ngSwitchCase="'endpoint'"
        id="{{ step.connection.name | synSlugify }}"
        class="text-center connection" 
        (click)="onViewDetails(step)" 
        title="{{ step.connection.name }}&nbsp;{{ step.action.name }}">
        <div [class]="'step-line ' + getStepLineClass(index)"></div>
        <div class="icon">
          <img class="syn-icon-small" [src]="step.connection | synIconPath">
        </div>
        <div class="syn-truncate__ellipsis">{{ step.connection.name | capitalize }}</div>
      </div>
      <div *ngSwitchDefault id="{{ (step.name || step.stepKind) | synSlugify }}" class="text-center step">
        <div class="step-line"></div>
        <div class="icon not-connection">
          <div class="icon-line"></div>
        </div>
        <div>{{ stepStore.getStepName(step) }}</div>
      </div>
    </ng-container>
    <span *ngIf="!last" class="step-sep fa fa-angle-right"></span>
  </div>
```

Please note how `*ngFor` or `*ngSwitchCase` have been applied directly on HTML elements, reducing the overall template boilerplate. ;)

Reply:
Thanks! Would I still need `<div *ngIf="integration.steps">` wrapping it?

Reply:
I kinda feel like it's a safer bet but technically I think the `*ngFor` shouldn't execute if `steps` is undefined.  So you can probably leave it out.

But as you can see we like to make no assumptions as to what we get back for an API request :-)

Reply:
thanks guys! updated the branch.